static/src/js/components/AccessMethod/SwodlrForm.jsx

```diff
@@ -0,0 +1,439 @@
+  geoRasterOptions
+} from '../../constants/swodlrConstants'
+
+import EDSCIcon from '../EDSCIcon/EDSCIcon'
+import ProjectPanelSection from '../ProjectPanels/ProjectPanelSection'
+
+import './SwodlrForm.scss'
+
+/**
+   * Renders Swodlr Form.
+   * @param {Object} props - The props passed into the component.
+   * @param {Array} props.granuleList - The list of selected granules for the current collection.
+   * @param {String} props.collectionId - The project collection id.
+   * @param {Function} props.onUpdateAccessMethod - Function to update metadata for the currently selected access method.
+   * @param {String} props.selectedAccessMethod - The currently seletected access method.
+   * @param {Function} props.setGranuleList - Function to update the granule list of the parent component.
+  */
+
+const SwodlrForm = ({
+  granuleList,
+  collectionId,
+  onUpdateAccessMethod,
+  selectedAccessMethod,
+  setGranuleList
+}) => {
+  const [granuleExtent, setGranuleExtent] = useState(false)
+  const [sampleGrid, setSampleGrid] = useState('UTM')
+  const [rasterResolution, setRasterResolution] = useState(90)
+
+  const handleSwoldrOptions = () => {
+    const customParams = {}
+
+    granuleList.forEach((granule) => {
+      const { id } = granule
+      customParams[id] = {}
+      if (sampleGrid === 'UTM') {
+        customParams[id].utmZoneAdjust = granule.utmZoneAdjust ? granule.utmZoneAdjust : 0
+        customParams[id].mgrsBandAdjust = granule.mgrsBandAdjust ? granule.mgrsBandAdjust : 0
+      } else {
+        customParams[id].utmZoneAdjust = null
+        customParams[id].mgrsBandAdjust = null
+      }
+    })
+
+    onUpdateAccessMethod({
+      collectionId,
+      method: {
+        [selectedAccessMethod]: {
+          swodlrData: {
+            params: {
+              rasterResolution,
+              outputSamplingGridType: sampleGrid,
+              outputGranuleExtentFlag: granuleExtent
+            },
+            custom_params: customParams
+          }
+        }
+      }
+    })
+  }
+
+  const handleCollectionGranuleListUpdate = (indexVal, property, e) => {
+    const granuleListCopy = granuleList
+
+    if (property === 'utm') {
+      granuleListCopy[indexVal].utmZoneAdjust = Number(e.target.value)
+    }
+
+    if (property === 'mgrs') {
+      granuleListCopy[indexVal].mgrsBandAdjust = Number(e.target.value)
+    }
+
+    setGranuleList(granuleList)
+  }
+
+  const handleRasterResolutionUpdate = (event) => {
+    setRasterResolution(Number(event.target.value))
+  }
+
+  const handleSampleGrid = (type) => {
+    setSampleGrid(type)
+    let defaultRasterValue
+    if (type === 'GEO') {
+      defaultRasterValue = 3
+    } else {
+      defaultRasterValue = 90
+    }
+
+    setRasterResolution(defaultRasterValue)
+  }
+
+  const handleGranuleExtent = (value) => {
+    setGranuleExtent(value)
+  }
+
+  useEffect(() => {
+    handleSwoldrOptions()
+  }, [granuleExtent, granuleList, rasterResolution, sampleGrid])
```

Comment:
Because `handleSwoldrOptions` is a separate function, it is unclear why each of these values are in the dependency array. That could be avoided by either not using a separate method, or passing the dependent values in as parameters to the function

Reply:
I instead added a comment. Since the values were state values, it didn't make sense to add them as parameters, and I am personally against doing the self invoking function thing inside an useEffect. I felt a comment here to explain what was happening was good as well.

Reply:
This causes over-rendering of the form and crashes EDSC

Reply:
That re-rending issue was coming from `granuleList` specifically, but, the others can be tracked in the dependency array